functional_tui_test.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+set -e
+
+echo "Cleaning up potentially lingering sessions"
+tmux kill-session -t server 2>/dev/null || true
+tmux kill-session -t client1 2>/dev/null || true
+tmux kill-session -t client2 2>/dev/null || true
+
+# Function to start a tmux session
+start_tmux_session() {
+    local name=$1
+    local command=$2
+    local log_file="${name}_output.log"
+    
+    # Clear existing log file
+    > "$log_file"
+    
+    # Start the tmux session with logging
+    tmux new-session -d -s $name
+    tmux pipe-pane -t $name "cat >> $log_file"
+    tmux send-keys -t $name "$command" C-m
+    
+    echo "Started $name in tmux session (Logging to $log_file)"
+}
+
+# Start the server
+echo "Starting server..."
+start_tmux_session "server" "cargo r -p server"
```

Comment:
If you built the binary first and then referenced the path to it you could avoid this long timeout.